### PR TITLE
Switch to support annotations on public APIs where appropriate

### LINKED ISSUE #1384: More consistent null input checks
Some methods accept null when it's a clear programming error to do so, and they fail silently rather than alert the developer that they're doing something wrong. Many of these methods correspond to other methods (e.g. inverse methods with the same input) that do perform null checks, so making the null checks consistent across all related methods would be ideal. There are other methods that fail on null input when their corresponding methods do just fine with a null input, often falling back to some reasonable default behavior (like showing a placeholder image).

- `Picasso.cancelRequest`: `ImageView`, `Target`, and `RemoteViews` variants. Corresponding methods `RequestCreator.into` all perform null checks, and providing null here is a programming error, so I believe null checks should throw on all these methods.
- `Picasso.cancelTag`, `pauseTag`, `resumeTag`: corresponding method `RequestCreator.tag` performs a null check, and a null tag doesn't make sense, so I believe null checks should throw on all these methods.
- `Picasso.invalidate`: `Uri`, `String`, and `File` variants. Corresponding methods `Picasso.load` all allow null, though the `String` variant fails on a blank/empty input. I'd argue that the `String` variant is commonly used with input coming from an external source like a backend server and therefore should be more permissive - null/empty/blank strings all seem reasonable, especially since there's such a clear and easy fallback to show the placeholder image. `Uri` and `File` objects are both created on the client side so it seems reasonable to reject null in those cases and allow those with invalid paths internally to proceed to the usual fallback placeholder/error behavior. Whatever is decided for `load` should be duplicated in the corresponding `invalidate` method. I believe that the `String` versions should allow null/empty/blank, while the `Uri` and `File` versions should throw on null.

### PR DESCRIPTION
**Pull request by hzsweers**
Also should resolve #1384 and removes the intellijAnnotations

### CODE REVIEW

File: picasso/src/main/java/com/squareup/picasso/Downloader.java [hunk 1]
```diff
@@ -32,7 +35,7 @@
    * loading the bitmap.
    * @throws IOException if the requested URL cannot successfully be loaded.
    */
-  Response load(Uri uri, int networkPolicy) throws IOException;
+  @Nullable Response load(@NonNull Uri uri, int networkPolicy) throws IOException;
```
Comment (JakeWharton) [inline picasso/src/main/java/com/squareup/picasso/Downloader.java]: We allow `null`? I thought it was `Response` or throw only...

File: picasso/src/main/java/com/squareup/picasso/Picasso.java [hunk 2]
```diff
@@ -34,6 +39,7 @@
 import java.util.WeakHashMap;
 import java.util.concurrent.ExecutorService;
 
+import static android.R.attr.tag;
```
Comment (JakeWharton) [inline picasso/src/main/java/com/squareup/picasso/Picasso.java]: Delete

File: picasso/src/main/java/com/squareup/picasso/Downloader.java (hunk 1 above)
Comment (hzsweers) [inline picasso/src/main/java/com/squareup/picasso/Downloader.java]: ¯\_(ツ)_/¯ I was just going by what the doc on line 34 said

File: picasso/src/main/java/com/squareup/picasso/Picasso.java (hunk 2 above)
Comment (hzsweers) [inline picasso/src/main/java/com/squareup/picasso/Picasso.java]: :O intelliJ 16's new auto-import-static

File: picasso/src/main/java/com/squareup/picasso/Request.java [hunk 3]
```diff
@@ -411,13 +416,16 @@ public Builder purgeable() {
     }
 
     /** Decode the image using the specified config. */
-    public Builder config(Bitmap.Config config) {
+    public Builder config(@NonNull Bitmap.Config config) {
+      if (config == null) {
+        throw new IllegalArgumentException("tag == null");
```
Comment (JakeWharton) [inline picasso/src/main/java/com/squareup/picasso/Request.java]: wrong message

File: picasso/src/main/java/com/squareup/picasso/Picasso.java (hunk 2 above)
Comment (hzsweers) [inline picasso/src/main/java/com/squareup/picasso/Picasso.java]: ce01e2d

File: picasso/src/main/java/com/squareup/picasso/Request.java (hunk 3 above)
Comment (hzsweers) [inline picasso/src/main/java/com/squareup/picasso/Request.java]: c88c020

File: picasso/src/main/java/com/squareup/picasso/Downloader.java (hunk 1 above)
Comment (JakeWharton) [inline picasso/src/main/java/com/squareup/picasso/Downloader.java]: Hmm... maybe returning null doesn't retry?

File: picasso/src/main/java/com/squareup/picasso/Picasso.java [hunk 4]
```diff
@@ -365,7 +376,7 @@ public RequestCreator load(int resourceId) {
    * @see #invalidate(String)
    * @see #invalidate(File)
    */
-  public void invalidate(Uri uri) {
+  public void invalidate(@NonNull Uri uri) {
```
Comment (JakeWharton) [inline picasso/src/main/java/com/squareup/picasso/Picasso.java]: This should be nullable since `into(Uri)` is nullable.

Comment (JakeWharton) [inline picasso/src/main/java/com/squareup/picasso/Picasso.java]: And obviously the check removed

File: picasso/src/main/java/com/squareup/picasso/Picasso.java [hunk 5]
```diff
@@ -379,9 +390,9 @@ public void invalidate(Uri uri) {
    * @see #invalidate(Uri)
    * @see #invalidate(File)
    */
-  public void invalidate(String path) {
+  public void invalidate(@Nullable String path) {
     if (path == null) {
-      throw new IllegalArgumentException("path == null");
+      return;
```
Comment (JakeWharton) [inline picasso/src/main/java/com/squareup/picasso/Picasso.java]: Flip the `if` and move the invalidate call up

Comment (hzsweers) [inline picasso/src/main/java/com/squareup/picasso/Picasso.java]: c3d2b2a

File: picasso/src/main/java/com/squareup/picasso/Picasso.java (hunk 4 above)
Comment (hzsweers) [inline picasso/src/main/java/com/squareup/picasso/Picasso.java]: 0106bef

**Pull request merged**